pkg/sql/type_change.go

```diff
@@ -902,6 +949,111 @@ func (t *typeSchemaChanger) canRemoveEnumValue(
 	return t.canRemoveEnumValueFromArrayUsages(ctx, arrayTypeDesc, member, txn, descsCol)
 }
 
+func findUsagesOfEnumValueInPartitioning(
+	partitioning catalog.Partitioning,
+	codec keys.SQLCodec,
+	columns []catalog.Column,
+	table catalog.TableDescriptor,
+	index catalog.Index,
+	member *descpb.TypeDescriptor_EnumMember,
+	fakePrefixDatums []tree.Datum,
+	typ *typedesc.Mutable,
+) (foundUsage bool, _ error) {
+	if partitioning == nil || partitioning.NumColumns() == 0 {
+		return false, nil
+	}
+
+	var colsToCheck util.FastIntSet
+	for i, c := range columns[:partitioning.NumColumns()] {
+		typT := c.GetType()
+		if !typT.UserDefined() {
+			continue
+		}
+		id, err := typedesc.UserDefinedTypeOIDToID(typT.Oid())
+		if err != nil {
+			return false, errors.WithAssertionFailure(err)
+		}
+		if id != typ.GetID() {
+			continue
+		}
+		colsToCheck.Add(i)
+	}
+	makeFakeDatums := func(n int) []tree.Datum {
```

Comment:
can you add a comment about why these fake datums are needed?

Reply:
Done.